### PR TITLE
filter out empty values in log fields

### PR DESCRIPTION
**Pull request by samjsong**
Since our backend cannot accept empty values for logs metadata, we filter out any empty values.

### CODE REVIEW

**Comment by samjsong** [comment]
Hmm I'm seeing failures with our build script, will take a look

File: Dockerfile [hunk 1]
```diff
@@ -71,6 +71,4 @@ ENV FORWARD_INPUT_PORT "24224"
 COPY --from=builder /usr/local/bundle /usr/local/bundle
 COPY ./entrypoint.sh /fluentd/
 
-USER fluent
```
[the remark this answers is not in the document]
Comment (samjsong) [reply Dockerfile]: We found that mounting the directory `/mnt/pos/` will actually override any tinkering we do with the permissions in the Dockerfile. So we'd have to use an init container or maybe set a security context in order to make this change. That, or run as root.

Going to deprioritize this and instead see if we can get the desired fix for empty/null label values in a different plugin in the new collection pipeline instead.

[the remark this answers is not in the document]
Comment (rvmiller89) [reply Dockerfile]: wow :(

**Comment by samjsong** [comment]
Closing for now for reasons above, can reopen in the future when we get back to it

**Pull request closed without merging**